auto_ncpt.ipynb

```diff
@@ -38,56 +38,505 @@
+    "            button_style=('success' if keep else ''),\n",
+    "        )\n",
+    "    \n",
+    "        delete_button = Button(\n",
+    "            tooltip=\"Delete this model\",\n",
+    "            button_style='danger',     \n",
+    "            layout=Layout(width='30px'),\n",
+    "            icon='trash-o' \n",
+    "        )\n",
+    "        \n",
+    "        def on_toggle(change):\n",
+    "            self.update_link_status(url,toggle_enable.value,toggle_keep.value)\n",
+    "            toggle_keep.button_style = 'success' if toggle_keep.value else ''\n",
+    "            toggle_enable.button_style = 'success' if toggle_enable.value else ''\n",
+    "            self.update_accordion_title(cat)\n",
+    "            self.save()\n",
+    "        \n",
+    "        def delete_action(b):\n",
+    "            self.update_link_status(url, False, False)\n",
+    "            self.remove_link(cat,url)\n",
+    "            self.save()\n",
+    "\n",
+    "        toggle_enable.observe(on_toggle, 'value')\n",
+    "        toggle_keep.observe(on_toggle, 'value')\n",
+    "        delete_button.on_click(delete_action)\n",
+    "        return toggle_enable , toggle_keep , delete_button\n",
+    "\n",
+    "    # create add ressource ui\n",
+    "    def create_add_link(self, category):\n",
+    "        link_control = Text(\n",
+    "            value=\"\",\n",
+    "            placeholder=category + \" link...\",\n",
+    "            layout=Layout(width='80%'),\n",
+    "        )\n",
+    "        \n",
+    "        add_button = Button(\n",
+    "            description='Add',\n",
+    "            layout=Layout(width='20%'),\n",
+    "            icon='plus' \n",
+    "        )\n",
+    "    \n",
+    "        def add_action(b):\n",
+    "            print(\"adding\")\n",
+    "            self.add_link(link_control.value,category )\n",
+    "            self.save()\n",
+    "\n",
+    "        add_button.on_click(add_action)\n",
+    "        return HBox([link_control, add_button])\n",
+    "        \n",
+    "    # creating output zip/clean tools \n",
+    "    def create_zip_tool(self):\n",
+    "        #view\n",
+    "        zip_button = Button(\n",
+    "            description='Zip',\n",
+    "            tooltip='Zip folder',\n",
+    "            icon='zipper' \n",
+    "        )\n",
+    "        clean_button = Button(\n",
+    "            description='Clean',\n",
+    "            tooltip='Cleaning folder',\n",
+    "            icon='zipper'\n",
+    "        )\n",
+    "        zip_clean_button = Button(\n",
+    "            description='Zip & Clean',\n",
+    "            button_style='info', \n",
+    "            tooltip='Zip folder and clean it up',\n",
+    "            icon='file' \n",
+    "        )\n",
+    "    \n",
+    "        #actions\n",
+    "        def clean_outputs(b):\n",
+    "            !rm -rf {b}\n",
+    "            b.mkdir(exist_ok=True)\n",
+    "            \n",
```

Comment:
this method only used once and it's just 2 lines of code
maybe write it to [line 348](https://github.com/NoCrypt/paperspace-ncpt/pull/3/files#diff-5c0d7cf5ba0ad0bcb5fa6658592061099f7aeed8e1ccf9db62cb899d9465fcdbR348)

Reply:
@RannaAI